src/common/consumer/consumer.c

```diff
@@ -3890,6 +3900,54 @@ unsigned long consumer_get_consume_start_pos(unsigned long consumed_pos,
 	return start_pos;
 }
 
+/*
+ * Check if a stream is ready to be rotated after extracting it.
+ *
+ * Return 1 if it is ready for rotation, 0 if it is not, a negative value on
+ * error. Stream lock must be held.
+ */
+int lttng_consumer_stream_is_rotate_ready(struct lttng_consumer_stream *stream)
+{
+	int ret;
+	unsigned long consumed_pos;
+
+	if (!stream->rotate_position && !stream->rotate_ready) {
+		ret = 0;
+		goto end;
+	}
+
+	if (stream->rotate_ready) {
+		ret = 1;
+		goto end;
+	}
+
+	/*
+	 * If we don't have the rotate_ready flag, check the consumed position
+	 * to determine if we need to rotate.
+	 */
+	ret = lttng_consumer_sample_snapshot_positions(stream);
+	if (ret < 0) {
+		ERR("Taking kernel snapshot positions");
```

Comment:
This function will be called by ust consumers as well as kernel consumers. We should remove the kernel mention.